portfolio/src/main/webapp/script.js

```diff
@@ -37,4 +37,10 @@ function autoChange (){
     document.getElementById("headshot").style.backgroundImage= headshots[headshotIndex];
 }
 
-setInterval(autoChange,5000)
\ No newline at end of file
+setInterval(autoChange,5000)
+
+function getName() {
+  fetch('/data').then(response => response.text()).then((name) => {
```

Comment:
Chaining promise `.then`s doesn't chain the output, it actually just attaches more listeners: https://stackoverflow.com/a/25346201. You've tried running this code? I would expect you'd see the whole response written to the `#NAME` element.

Reply:
Same comment as on Nina's, fetch's response object is special. .text() returns a promise I believe

Reply:
Ah nvm, looked into this a bit more and looks like I'm incorrect here. All the same, I think it might be more readable to condense these into the same `then`.

Reply:
@Tetheta It looks like this behavior is due to `.then` always returning a promise with the inner return value: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

Reply:
Not the case here, note this: https://developer.mozilla.org/en-US/docs/Web/API/Body/text

I've used this API in the past, it's weird but you have to .then the .text property, it's async.

Reply:
Ahhh TIL : ) ignore my comment then

Reply:
I ran it and only the name showed up. I just did what the walk-through showed.

Reply:
@LukeGMTravis Yeah I had the same reaction when I first saw it. Definitely does not follow best practice naming =/ 

@hailed748  This type of little comment chain is pretty common on many Google CLs internally too haha. So guess this is a good taste of that :P 
Also a good point that us reviewers aren't always right. Feel free to link walkthrough code when you get pushback on something they told you to do. 
Sometimes we just haven't used the same APIs (fetch has a few issues for big company use), or sometimes they're taking shortcuts you wouldn't normally in prod code.